### PR TITLE
[KYUUBI #5935] Engine YARN mode support proxy appUser

### PR DESCRIPTION
**Pull request by Yikf**
# :mag: Description
## Issue References 🔗

This pull request fixes https://github.com/apache/kyuubi/issues/5935

## Describe Your Solution 🔧

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Types of changes :bookmark:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Related Unit Tests

---

# Checklist 📝

- [x] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

**Be nice. Be informative.**

### CODE REVIEW

**Comment by Yikf** [comment]
image

File: kyuubi-common/src/main/scala/org/apache/kyuubi/engine/deploy/yarn/EngineYarnModeSubmitter.scala [hunk 1]
```diff
@@ -92,6 +94,28 @@ abstract class EngineYarnModeSubmitter extends Logging {
   def engineExtraJars(): Seq[File] = Seq.empty
 
   protected def submitApplication(): Unit = {
+    val user = kyuubiConf.getOption(KYUUBI_SESSION_USER_KEY)
```
Comment (pan3793) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/engine/deploy/yarn/EngineYarnModeSubmitter.scala:97]: we actually have seen that limitation of always using session user as the engine launch user, let's define a dedicated `proxyUser: Option[String]`

**Comment by pan3793** [comment]
BTW, for hadoop-testing environment, the recommended `HIVE_HADOOP_CLASSPATH` is `/opt/hadoop/share/hadoop/client/*`, there are only Hadoop Shaded client jars

File: kyuubi-common/src/main/scala/org/apache/kyuubi/engine/deploy/yarn/EngineYarnModeSubmitter.scala (hunk 1 above)
Comment (Yikf) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/engine/deploy/yarn/EngineYarnModeSubmitter.scala:97]: The user retrieved in this context is identified as appUser, which is supplied via the ProcessBuilder using the --conf parameter. However, the naming of the configuration key associated with this user appears to be somewhat unclear. 

Are you suggesting that the key name should be updated for clarity, or are you indicating that appUser does not represent the intended proxy user?

**Pull request closed without merging**